wgsl/index.bs

```diff
@@ -4180,6 +4180,242 @@ TODO: deduplicate these tables
 
 ## Texture built-in functions ## {#texture-builtin-functions}
 
+### `textureSample` ### {#texturesample}
+
+Samples a texture.
+
+```rust
+textureSample(texture : texture_sampled_1d<type>,         sampler : sampler, coords : f32                                             ) -> <type>
+textureSample(texture : texture_sampled_1d<type>,         sampler : sampler, coords : f32,                          offset : i32      ) -> <type>
```

Comment:
Note that Metal limits the "offset" support only to 2D, 2D array, and 3D textures, so that should reduce our overload set as well

Reply:
Done